libs/auth/react/src/lib/Authenticator/CheckIdpSession.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { SyntheticEvent, useCallback, useReducer, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import { getAuthSettings, getUserManager } from '../userManager'
+import { ActionType, initialState, reducer } from './Authenticator.state'
+
+import type { History } from 'history'
+import { AuthSettings } from '../AuthSettings'
+
+export interface SessionInfo {
+  // Message detailing if the request was processed OK, no session detected or with failure.
+  message: string
+
+  // The time when the authenticated session expires.
+  expiresUtc?: string
+
+  // Boolean flag to indicated if the Expires time is passed.
+  isExpired?: boolean
+}
+
+const messageEventName = 'message'
+
+const getReturnUrl = (history: History, { redirectPath }: AuthSettings) => {
+  const returnUrl = history.location.pathname + history.location.search
+  if (redirectPath && returnUrl.startsWith(redirectPath)) {
+    return '/'
+  }
+  return returnUrl
+}
+
+export const CheckIdpSession = () => {
+  const history = useHistory()
+  const reducerInstance = useReducer(reducer, initialState)
+  const [state, dispatch] = reducerInstance
+  const userManager = getUserManager()
+  const authSettings = getAuthSettings()
+  const iframeSrc = `${authSettings.authority}${authSettings.checkSessionPath}`
+  const [iframeChecksum, setIframeChecksum] = useState(0)
+
+  let sessionTimeout: ReturnType<typeof setTimeout> | null = null
+
+  const signIn = useCallback(
+    async function signIn() {
+      dispatch({
+        type: ActionType.SIGNIN_START,
+      })
+      return userManager.signinRedirect({
+        state: getReturnUrl(history, authSettings),
+      })
+      // Nothing more happens here since browser will redirect to IDS.
+    },
+    [dispatch, userManager, authSettings, history],
+  )
+
+  const onLoadHandler = (event: SyntheticEvent<HTMLIFrameElement>) => {
+    console.log('***** iFrame loaded', event)
```

Comment:
debug.

Reply:
Thought I had cleaned up all my console logs ...  apparently not 😅